airgun/widgets.py

```diff
@@ -1321,6 +1321,42 @@ class EditableEntryMultiCheckbox(EditableEntry):
     edit_field = CheckboxGroup(locator='.//form')
 
 
+class EditableEntryMultiTextInput(GenericLocatorWidget):
+    """
+    A set of text inputs
+    """
+    ITEMS_LOCATOR = './/label'
+    TEXTINPUT_LOCATOR = './/label[normalize-space(.)="{}"]/following-sibling::div/textarea'
+
+    @cached_property
+    def textinputs(self):
+        labels = [
+            line.text
+            for line in self.browser.elements(self.ITEMS_LOCATOR, parent=self)
+        ]
```

Comment:
think labels can be as a standalone cached property

Reply:
I don't understand what you are suggesting.

Reply:
@ldjebran

Reply:
I believe @ldjebran means to create `labels` property. Something like that:

```python
    @cached_property
    def labels(self):
        return [
            line.text
            for line in self.browser.elements(self.ITEMS_LOCATOR, parent=self)
        ]

    @cached_property
    def textinputs(self):
        return {
            label: TextInput(self, locator=self.TEXTINPUT_LOCATOR.format(label))
            for label in self.labels
        }
```